e2e/compatibility/consul_test.go

```diff
@@ -537,6 +577,24 @@ func registerCatalog(t *testing.T, entity *capi.CatalogRegistration, port int) {
 	time.Sleep(8 * time.Second)
 }
 
+// getCheck returns a Consul health check if it is registered
+func getCheck(t *testing.T, checkID string, port int) (*capi.AgentCheck, error) {
+	conf := capi.DefaultConfig()
+	conf.Address = fmt.Sprintf("localhost:%d", port)
+	client, err := capi.NewClient(conf)
+	require.NoError(t, err)
+
+	checks, err := client.Agent().Checks()
+	if err != nil {
+		return nil, err
+	}
+	c, ok := checks[checkID]
+	if !ok {
```

Comment:
I noticed that when using getCheck and getService, the caller requires no error. Should we just use the passed in *testing.T to require true?
```Go
require.True(t, ok)
```

Reply:
I was leaving it to the caller for the future if we need to verify the opposite, where the desired behavior is that a check or service doesn't exist.

But I probably should have done an error assertion right before this in the case where getting the agent checks failed! We should always expect to get a list of checks back.